<?xml version="1.0" ?>
<!-- Revolute-Revolute Manipulator -->
<robot name="simple_rgbd_cam" xmlns:xacro="http://www.ros.org/wiki/xacro">

  <!-- Constants for robot dimensions -->
  <xacro:property name="PI" value="3.1415926535897931" />
  <xacro:property name="mass" value="1" /> <!-- arbitrary value for mass -->
  <xacro:property name="width" value="0.064" /> <!-- Square dimensions (widthxwidth) of beams -->
  <xacro:property name="height" value="0.048" /> <!-- Link 1 -->
  <xacro:property name="depth" value="0.1" /> <!-- Link 1 -->
  <xacro:property name="camera_link" value="0.01" /> <!-- Size of square 'camera' box -->

  <!-- Import all Gazebo-customization elements, including Gazebo colors -->
  <xacro:include filename="$(find extrinsics_calibrator)/urdf/simple_rgbd_cam.gazebo" />
  <!-- Import Rviz colors -->
  <xacro:include filename="$(find extrinsics_calibrator)/urdf/materials.xacro" />

  <!-- Used for fixing robot to Gazebo 'base_link' -->
  <!--   <link name="world"/> -->

  <!--   <joint name="fixed" type="fixed">
    <parent link="world"/>
    <child link="link1"/>
  </joint> -->

  <!-- Base Link -->
  <link name="link1">
    <static>true</static>
    <collision>
      <origin xyz="0 0 -${depth/2}" rpy="0 0 0" />
      <geometry>
        <box size="${width} ${height} ${depth}" />
      </geometry>
    </collision>

    <visual>
      <origin xyz="0 0 -${depth/2}" rpy="0 0 0" />
      <geometry>
        <box size="${width} ${height} ${depth}" />
      </geometry>
      <material name="orange" />
    </visual>


  </link>

  <joint name="camera_joint" type="fixed">
    <origin xyz="0 0 0" rpy="0 -${pi/2} ${pi/2}" />
    <parent link="link1" />
    <child link="camera_link" />
  </joint>

  <!-- Camera -->
  <link name="camera_link">
    <collision>
      <origin xyz="0 0 ${camera_link/2}" rpy="0 0 0" />
      <geometry>
        <box size="${camera_link} ${camera_link} ${camera_link}" />
      </geometry>
    </collision>

    <visual>
      <origin xyz="0 0 ${camera_link/2}" rpy="0 0 0" />
      <geometry>
        <box size="${camera_link} ${camera_link} ${camera_link}" />
      </geometry>
      <material name="red" />
    </visual>

    <inertial>
      <mass value="1e-5" />
      <origin xyz="0 0 0" rpy="0 0 0" />
      <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6" />
    </inertial>
  </link>

  <!-- generate an optical frame http://www.ros.org/reps/rep-0103.html#suffix-frames
      so that ros and opencv can operate on the camera frame correctly -->
  <joint name="camera_optical_joint" type="fixed">
    <!-- these values have to be these values otherwise the gazebo camera image
        won't be aligned properly with the frame it is supposedly originating from -->
    <!-- <origin xyz="0 0 0" rpy="${-pi/2} 0 ${-pi/2}" /> -->
    <origin xyz="0 0 0" rpy="0 0 0" />
    <parent link="camera_link" />
    <child link="camera_link_optical" />
  </joint>

  <link name="camera_link_optical"></link>

</robot>